src/clib/pio_darray_int.c

```diff
@@ -1425,6 +1446,309 @@ int pio_read_darray_nc_serial(file_desc_t *file, int fndims, io_desc_t *iodesc,
+            preqs && nreqs && last_var_with_req &&
+            preq_block_ranges && nreq_blocks);
+    assert(file->iotype == PIO_IOTYPE_PNETCDF);
+    assert(file->iosystem && (file->iosystem->num_iotasks > 0));
+
+    *nreqs = 0;
+    *nvars_with_reqs = 0;
+    *last_var_with_req = 0;
+    *nreq_blocks = 0;
+
+    int file_nreqs = 0;
+    int vdesc_with_reqs_start = 0;
+    int vdesc_with_reqs_end = 0;
+    /* FIXME: Update file->nreqs with vdesc->nreqs to avoid computing
+     * everytime
+     */
+    for(int i = 0; i < PIO_MAX_VARS; i++){
+      var_desc_t *vdesc = file->varlist + i;
+      if(vdesc->nreqs > 0){
+        file_nreqs += vdesc->nreqs;
+        /* Once the range starts, vdesc_with_reqs_end >= 1 */
+        if(vdesc_with_reqs_end == 0){
+          vdesc_with_reqs_start = i;
+        }
+        vdesc_with_reqs_end = i + 1;
+        (*nvars_with_reqs)++;
+        *last_var_with_req = i;
+      }
+    }
+
+#ifdef PIO_ENABLE_SANITY_CHECKS
+    /* Sanity check : All io ranks have the same number of reqs per file */
+    int file_nreqs_root = file_nreqs;
+    mpierr = MPI_Bcast(&file_nreqs_root, 1, MPI_INT,
+              file->iosystem->ioroot, file->iosystem->io_comm);
+    assert(mpierr == MPI_SUCCESS);
+    assert(file_nreqs_root == file_nreqs);
+#endif
+
+    /* No requests pending on this file */
+    if(file_nreqs == 0){
+      *nreqs = 0;
+      *nreq_blocks = 0;
+
+      return PIO_NOERR;
+    }
+
+    /* preqs : pointer to consolidated list of pending requests on this file */
+    *preqs = (int *) calloc(file_nreqs, sizeof(int));
+    if(!(*preqs)){
+      return pio_err(file->iosystem, file, PIO_ENOMEM, __FILE__, __LINE__,
+                      "Unable to allocate memory (%lld bytes) for consolidating pending requests on a file (%s, ncid=%d, num pending requests = %d)", (unsigned long long) (file_nreqs * sizeof(int)), pio_get_fname_from_file(file), file->pio_ncid, file_nreqs);
+    }
+    *nreqs = file_nreqs;
+
+    /* preq_block_ranges : Contains the ranges for the blocks,
+     * each block of size <= file_req_block_sz_limit, which includes
+     * start indices in the first half and end indices in the second half.
+     * The number of block ranges should be less than the number of
+     * pending requests (max number of blocks => 1 request per block)
+     * => 2 * file_nreqs
+     * This buffer is also used to store the size of block ranges at
+     * index 2 * file_nreqs => 1 int (for communicating with non-root procs)
+     * => size of 2 * file_nreqs + 1
+     */
+    *preq_block_ranges = (int *) calloc(2 * file_nreqs + 1, sizeof(int));
+    if(!(*preq_block_ranges)){
+      return pio_err(file->iosystem, file, PIO_ENOMEM, __FILE__, __LINE__,
+                      "Unable to allocate memory (%lld bytes) for storing pending request ranges in a file (%s, ncid=%d, num pending requests = %d)", (unsigned long long) ((2 * file_nreqs + 1)* sizeof(int)), pio_get_fname_from_file(file), file->pio_ncid, file_nreqs);
```

Comment:
For `unsigned long long`, consider using `%llu`?

Reply:
ok, will change and rebase

Reply:
Make sure to change all occurrences, thanks.

Reply:
I fixed the two instances in this PR